### PR TITLE
OS-8010 move platform repositories out of Gerrit

### PR DESCRIPTION
**Pull request by jlevon**
Took the opportunity here to re-organize the readme a bit more sensibly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,37 +8,76 @@ gmake world
...
+
+* In the *#smartos* IRC channel on the [Freenode IRC
+  network](https://freenode.net/).
+
+# Bug Reports
+
+If you encounter a problem, please reach out for assistance. You can
+file a [github issue](https://github.com/joyent/smartos-live/issues) for
+any problem you encounter. When filing a bug, please include the
+platform version that you're running and a description of the problem.
```
Comment (trentm) [inline README.md]: Unrelated. If bug reports lacking info abound, we could consider and "issue template" for the repo: https://help.github.com/en/github/building-a-strong-community/about-issue-and-pull-request-templates

File: README.md [hunk 2]
```diff
@@ -10,37 +8,76 @@ gmake world
...
-2. [Building SmartOS](#building-smartos)
-   - [Incremental Development](#incremental-development)
-3. [Testing and Things to Think About](#testing-and-things-to-think-about)
-4. [Contributing](#contributing)
-5. [Community](#community)
-6. [Bug Reports](#bug-reports)
+While there are multiple repositories that make up the smartos-live
+image, if you're in doubt about where to file a bug or just are
+uncertain, please file it on the [SmartOS live issue
+tracker](https://github.com/joyent/smartos-live) and we'll help from
```
Comment (trentm) [inline README.md]:
```suggestion
tracker](https://github.com/joyent/smartos-live/issues) and we'll help from
```

perhaps?

File: README.md [hunk 3]
```diff
@@ -684,7 +721,82 @@ usr/src/uts/intel/os/name_to_major
 usr/src/uts/intel/os/minor_perm
 ```
 
-## Testing and Things to Think About
+# Contributing
```
Comment (trentm) [inline README.md]: If you *like*, this section could be moved to a separate CONTRIBUTING.md file. GitHub will then provide an info box and a link to that file in the UI for creating a PR.

File: README.md [hunk 4]
```diff
@@ -684,7 +721,82 @@ usr/src/uts/intel/os/name_to_major
...
+new changes.
+
+All changes should have an associated issue. You can use the [GitHub
+issue tracker](https://github.com/joyent/smartos-live/issues). Joyent
+employees use an internal JIRA exposed at
+<https://smartos.org/bugview>. The commit message should be of the form used by
+illumos-gate, for example:
+
+```
+smartos-live#9999 make some changes
```
Comment (trentm) [inline README.md]: I don't have a preference at all, but my understanding was that we tended to include the "joyent/" prefix on the issue name. E.g.:

```
commit 0158021c21657bdc537895a634d1a73b593e6639
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2019-05-03T18:29:31-07:00 (6 months ago)

    joyent/smartos-live#771 imgadm vacuum will try to reap images used by bhyve vms
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>
```

Me testing GitHub automatic linking:
- without joyent prefix: smartos-live#771
- with joyent prefix: joyent/smartos-live#771

Comment (trentm) [reply README.md]: Yah, the "with prefix" form links up automatically, FWIW.

File: README.md [hunk 5]
```diff
@@ -684,7 +721,82 @@ usr/src/uts/intel/os/name_to_major
...
+that might only arise after prolonged use?
+* Is this a particularly risky change? If so, should I wait
+until the start of the next release cycle to put this back?
+* Are there any heads up notices I need to send as part of this? For
+example, this might happen because of a flag day.
+* Have I added a new tool that's required to run at build-time and
+tested this on older platform images?
+
+Prior to a PR being merged, it must have at least one code reviewer and one
+approver. They can be the same person, but two sets of eyes are preferred.
```
Comment (trentm) [inline README.md]: This should (eventually?) include a description of the "integration-approval" GitHub label process for getting an "approver"?

File: README.md [hunk 6]
```diff
@@ -887,10 +999,10 @@ failed, the names of those suites are emitted by `smartos-test` just before
 the script exits.
 
 When developers are adding tests to illumos, they should ensure that new
-tests are added to `$SRC/usr/src/pkg/manifests/*.mf` as these IPS
+tests are added to `$SRC/usr/src/pkg/manifests/\*.mf` as these IPS
```
Comment (trentm) [inline README.md]: Markdown syntax: FWIW, that escaping shouldn't be necessary in backticks.

File: README.md [hunk 7]
```diff
@@ -410,9 +447,9 @@ The `configure-projects` file takes the format:
 <path relative to ./projects>:<project branch>:[project git repo URL or path]
 ```
 
-The special tokens `cr` or `origin` can be used in place of a full git
-repo URL to denote either standard github.com or joyent gerrit URLs for that
-project. If no URL is given, we default to github.com.
+The special `origin` can be used in place of a full git repo URL to denote the
```
Comment (timfoster) [inline README.md]: "special token `origin`", or are we using 'special' as a noun here?

File: README.md [hunk 8]
```diff
@@ -684,7 +721,82 @@ usr/src/uts/intel/os/name_to_major
...
+<https://smartos.org/bugview>. The commit message should be of the form used by
+illumos-gate, for example:
+
+```
+smartos-live#9999 make some changes
+Reviewed-by: Steve Reviewer <steve.reviewer@gmail.com>
+Approved-by: Amy Approver <amy.approver@gmail.com>
+```
+
+(Yes, there is no blank line after the commit title.)
```
[the remark this answers is not in the document]
Comment (timfoster) [reply README.md]: I just checked - if you omit the title of the PR when merging, and move the title into the comments box, github will restore the title, leading to a duplicate. We still get a commit of the form:

```
timf@iorangi-eth0 (master) git log
commit eb09ee56cf6f20084bba6077e8ea08f2b629f201 (HEAD -> master, origin/master, origin/HEAD)
Author: Tim Foster <tim.foster@joyent.com>
Date:   Thu Nov 7 09:44:43 2019 +0000

    merge our conflicts again (#24)

    merge our conflicts again (#24)
    This is the 2nd line of the change
```

File: README.md (hunk 3 above)
Comment (jlevon) [reply README.md]: I don't think this is really necessary. Everyone needs to read the whole thing anyway.

File: README.md (hunk 4 above)
Comment (jlevon) [reply README.md]: Done, also updated this for the newer RFD175 format.

File: README.md (hunk 6 above)
Comment (jlevon) [reply README.md]: vim disagrees, and it doesn't hurt rendering on github, so it's nicer to edit with... yeah, it's probably a vim bug.

File: README.md (hunk 1 above)
Comment (jlevon) [reply README.md]: Maybe later yes.

File: README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (jlevon) [reply README.md]: Yep, I needed to update this for current RFD175

File: README.md [hunk 9]
```diff
@@ -723,22 +723,28 @@ usr/src/uts/intel/os/minor_perm
...
-illumos-gate, for example:
+<https://smartos.org/bugview>. The commit message should be of this form:
 
 ```
-smartos-live#9999 make some changes
-Reviewed-by: Steve Reviewer <steve.reviewer@gmail.com>
-Approved-by: Amy Approver <amy.approver@gmail.com>
+joyent/smartos-live#9999 make some changes (#23)
+
+joyent/smartos-live#10000
```
Comment (trentm) [inline README.md]: nit: Presumably issue 10000 has a title that should be included here?

**Pull request merged**